File: app/src/main/java/net/sunniwell/app/linktaro/nettv/view/CustomInfoDialog.java
package net.sunniwell.app.linktaro.nettv.view;

import android.app.Dialog;
import android.content.Context;
import android.view.KeyEvent;
import android.view.View;
import android.view.Window;
import android.view.WindowManager.LayoutParams;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.TextView;

public class CustomInfoDialog extends Dialog {
    private Button mBtnInfoCancel;
    private Context mContext;
    private boolean mCustomKey;
    private FrameLayout mFlResBg;
    private int mKeyCode;
    private TextView mTvResContent;

    public CustomInfoDialog(Context context, int theme) {
        super(context, theme);
        this.mContext = context;
        setContentView(C0395R.layout.info_dialog);
        initUi();
    }

    public CustomInfoDialog(Context context, int theme, boolean customKey, int keyCode) {
        super(context, theme);
        this.mContext = context;
        this.mCustomKey = customKey;
        this.mKeyCode = keyCode;
        setContentView(C0395R.layout.info_dialog);
        initUi();
    }

    private void initUi() {
        this.mFlResBg = (FrameLayout) findViewById(C0395R.C0397id.fl_infodialog);
        this.mTvResContent = (TextView) findViewById(C0395R.C0397id.tv_infodialog);
        setBtnCancelClickListener(null);
        Window dialogWindow = getWindow();
        LayoutParams lp = dialogWindow.getAttributes();
        dialogWindow.setGravity(49);
        lp.y = 200;
        dialogWindow.setAttributes(lp);
    }

    public void setDialogBg(int bgResource) {
        this.mFlResBg.setBackgroundResource(bgResource);
    }

    public void setTextContent(String text) {
        this.mTvResContent.setText(text);
    }

    public void setBtnCancelBgResource(int bgResource) {
        if (this.mBtnInfoCancel != null) {
            this.mBtnInfoCancel.setBackgroundResource(bgResource);
        }
    }

    private void setBtnCancelClickListener(OnClickListener listener) {
        if (this.mBtnInfoCancel != null) {
            this.mBtnInfoCancel.setOnClickListener(new OnClickListener() {
                public void onClick(View v) {
                    CustomInfoDialog.this.dismiss();
                }
            });
        }
    }

    public void show() {
        super.show();
        if (this.mBtnInfoCancel != null) {
            this.mBtnInfoCancel.requestFocus();
        }
    }

    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (!this.mCustomKey || keyCode != this.mKeyCode) {
            return super.onKeyDown(keyCode, event);
        }
        dismiss();
        return false;
    }
}

File: app/src/main/java/com/eliving/tv/customview/CustomInfoDialog.kt
package com.eliving.tv.customview

import android.app.Dialog
import android.content.Context
import android.view.KeyEvent
import android.view.View
import android.widget.Button
import android.widget.FrameLayout
import android.widget.TextView
import com.eliving.tv.R

class CustomInfoDialog : Dialog {
    private val mBtnInfoCancel: Button? = null
    private var mContext: Context
    private var mCustomKey = false
    private var mFlResBg: FrameLayout? = null
    private var mKeyCode = 0
    private var mTvResContent: TextView? = null

    constructor(context: Context, theme: Int) : super(context, theme) {
        mContext = context
        setContentView(R.layout.info_dialog)
        initUi()
    }

    constructor(
        context: Context,
        theme: Int,
        customKey: Boolean,
        keyCode: Int
    ) : super(context, theme) {
        mContext = context
        mCustomKey = customKey
        mKeyCode = keyCode
        setContentView(R.layout.info_dialog)
        initUi()
    }

    private fun initUi() {
        mFlResBg = findViewById<View>(R.id.fl_infodialog) as FrameLayout
        mTvResContent = findViewById<View>(R.id.tv_infodialog) as TextView
        setBtnCancelClickListener(null)
        val dialogWindow = window
        val lp = dialogWindow!!.attributes
        dialogWindow.setGravity(49)
        lp.y = 200
        dialogWindow.attributes = lp
    }

    fun setDialogBg(bgResource: Int) {
        mFlResBg!!.setBackgroundResource(bgResource)
    }

    fun setTextContent(text: String?) {
        mTvResContent!!.text = text
    }

    fun setBtnCancelBgResource(bgResource: Int) {
        if (mBtnInfoCancel != null) {
            mBtnInfoCancel.setBackgroundResource(bgResource)
        }
    }

    private fun setBtnCancelClickListener(listener: View.OnClickListener?) {
        if (mBtnInfoCancel != null) {
            mBtnInfoCancel.setOnClickListener(View.OnClickListener { dismiss() })
        }
    }

    override fun show() {
        super.show()
        if (mBtnInfoCancel != null) {
            mBtnInfoCancel.requestFocus()
        }
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (!mCustomKey || keyCode != mKeyCode) {
            return super.onKeyDown(keyCode, event)
        }
        dismiss()
        return false
    }
}
